api/controllers/console/wraps.py

```diff
@@ -419,3 +421,70 @@ def decorated(*args: P.args, **kwargs: P.kwargs):
         return view(*args, **kwargs)
 
     return decorated
+
+
+def _decrypt_field(field_name: str, error_class: type[Exception], error_message: str) -> None:
+    """
+    Helper to decode a Base64 encoded field in the request payload.
+
+    Args:
+        field_name: Name of the field to decode
+        error_class: Exception class to raise on decoding failure
+        error_message: Error message to include in the exception
+    """
+    if request and request.is_json:
+        # Get the payload dict - it's cached and mutable
+        payload = request.get_json()
+        if payload and field_name in payload:
+            encoded_value = payload[field_name]
+            decoded_value = FieldEncryption.decrypt_field(encoded_value)
+
+            # Update payload dict in-place with decoded value
+            # Since payload is a mutable dict and get_json() returns the cached reference,
+            # modifying it will affect all subsequent accesses including console_ns.payload
+            if decoded_value is not None:
+                payload[field_name] = decoded_value
+
+
+def decrypt_password_field(view: Callable[P, R]):
+    """
+    Decorator to decrypt password field in request payload.
+
+    Automatically decrypts the 'password' field if encryption is enabled.
+    If decryption fails, raises AuthenticationFailedError.
+
+    Usage:
+        @decrypt_password_field
+        def post(self):
+            args = LoginPayload.model_validate(console_ns.payload)
+            # args.password is now decrypted
+    """
+
+    @wraps(view)
+    def decorated(*args: P.args, **kwargs: P.kwargs):
+        _decrypt_field("password", AuthenticationFailedError, "Invalid encrypted data")
```

Comment:
If we decide to decode using this fixed string method, then it should be a constant.